beacon/blockchain/process_proposal.go

```diff
@@ -83,13 +83,26 @@ func (s *Service[
 		s.logger.Info("Received incoming blob sidecars")
 
 		// TODO: Clean this up once we remove generics.
-		c := convertConsensusSidecars[
+		cs := convertConsensusSidecars[
 			ConsensusSidecarsT,
 			BlobSidecarsT,
 		](consensusSidecars)
 
+		// Get the sidecar verification function from the state processor
+		//nolint:govet	// err shadowing
+		sidecarVerifierFn, err := s.stateProcessor.GetSidecarVerifierFn(
+			s.storageBackend.StateFromContext(ctx),
```

Comment:
q: here we verify sidecar first then block.
Does this matter? Would this verification fail potentially if we swapped order?
My guess is yes, since the block processing may change fork for instance.
If so, consider commenting on this.

Reply:
Another question: is this change backward compatible? Or should we introduce it past a certain fork.

Reply:
as discussed off-line, let's proceed with fixing this and handle differently backward compatibility issues, if any